tests/test_callbacks.py

```diff
@@ -103,6 +103,20 @@ def test_cpp_callable_cleanup():
     assert alive_counts == [0, 1, 2, 1, 2, 1, 0]
 
 
+def test_cpp_correct_overload_resolution():
```

Comment:
Can we also add a class test (that uses def __call__) here as well?

Reply:
Done